app/main.py

```diff
@@ -1,3 +1,29 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+rezult_all = []
+
+
+def cache(func: Callable) -> Callable:
+
+    def inner(*args, **kqargs) -> Any:
+        rezult = {"name": func, "arg": args}
+        if len(rezult_all) < 1:
+            rez_func = func(*args, **kqargs)
+            print("Calculating new result")
+            rezult["rezult"] = rez_func
+            rezult_all.append(rezult)
+            return rez_func
+
+        for rez in rezult_all:
+            if rez["name"] == rezult["name"] and rez["arg"] == rezult["arg"]:
+                print("Getting from cache")
+                return rez["rezult"]
+
+        print("Calculating new result")
+        rez_func = func(*args, **kqargs)
+        rezult["rezult"] = rez_func
+        rezult_all.append(rezult)
+        return rez_func
```

Comment:
U are making the same thing in 12-16 lines, use the DRY principle and make refactor of ur code